### PR TITLE
[DROOLS-6774] Implement interpreted (no-codegen) ClusteringModel

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @jiripetrlik 

**Thank you for submitting this pull request**

**NOTE!:** Double check the target branch for this PR.
The default is `main` so it will target Drools 8 / Kogito / Optaplanner 8.
If this PR is not strictly related to drools and kogito project in `drools.git`, it should probably target `7.x`as a branch

**Ports** If a forward-port or a backport is needed, paste the forward port PR here

[link](https://www.example.com)

**JIRA**:

[link](https://issues.redhat.com/browse/DROOLS-6774)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>

* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`

* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-api/src/main/java/org/kie/pmml/compiler/api/utils/ModelUtils.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.<
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-api/src/main/java/org/kie/pmml/compiler/api/utils/ModelUtils.java]: Probably a typo?

**Review by jiripetrlik — APPROVED** [review]

File: kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-api/src/main/java/org/kie/pmml/compiler/api/utils/ModelUtils.java (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-api/src/main/java/org/kie/pmml/compiler/api/utils/ModelUtils.java]: thanks @jiripetrlik :blush: !

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** `#341` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 453
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/341/testReport/org.kie.kogito.springboot.outbox/OutboxIT/testSendProcessEvents/">org.kie.kogito.springboot.outbox.OutboxIT.testSendProcessEvents</a></summary>
org.awaitility.core.ConditionTimeoutException: <br/>Assertion condition defined as a lambda expression in org.kie.kogito.springboot.outbox.OutboxIT 1 expectation failed.<br/>Expected status code <200> but was <404>.<br/> within 1 minutes.<br/>	at org.kie.kogito.springboot.outbox.OutboxIT.testSendProcessEvents(OutboxIT.java:129)<br/>Caused by: java.lang.AssertionError: <br/>1 expectation failed.<br/>Expected status code <200> but was <404>.<br/><br/>	at org.kie.kogito.springboot.outbox.OutboxIT.lambda$testSendProcessEvents$1(OutboxIT.java:136)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.downstream.kogito-examples/341/display/redirect

**Review by danielezonca — APPROVED** [review]

**Comment by gitgabrio** [comment]
HI @danielezonca 
could you please merge this ? Thanks!

**Pull request merged**